src/zenml/zen_stores/migrations/versions/1de87a8c8e8c_rename_project_to_workspace.py

```diff
@@ -0,0 +1,214 @@
+
+    Args:
+        source: Source table name.
+        constraint_name: Name of the foreign key constraint.
+    """
+    naming_convention = {
+        "fk": "fk_%(table_name)s_%(column_0_name)s_%(referred_table_name)s",
+    }
+    with op.batch_alter_table(
+        source, naming_convention=naming_convention
+    ) as batch_op:
+        batch_op.drop_constraint(
+            constraint_name=constraint_name,
+            type_="foreignkey",
+        )
+
+
+def _create_fk_constraint(
+    source: str,
+    target: str,
+    source_column: str,
+    target_column: str,
+    ondelete: str,
+) -> None:
+    """Create a foreign key constraint.
+
+    Args:
+        source: Source table name.
+        target: Target table name.
+        source_column: Source column name.
+        target_column: Target column name.
+        ondelete: On delete behavior.
+    """
+    constraint_name = _fk_constraint_name(source, target, source_column)
+    with op.batch_alter_table(source) as batch_op:
+        batch_op.create_foreign_key(
+            constraint_name=constraint_name,
+            referent_table=target,
+            local_cols=[source_column],
+            remote_cols=[target_column],
+            ondelete=ondelete,
+        )
+
+
+def _get_changes() -> Tuple[
+    List[str],
+    List[Tuple[str, str, str, str, str]],
+    List[Tuple[str, str, str, str, str]],
+]:
+    """Define the data that should be changed in the schema.
+
+    Returns:
+        A tuple of four lists:
+        - table names
+        - old foreign key constraints:
+            (source, target, source_column, target_column, ondelete)
+        - new foreign key constraints
+            (source, target, source_column, target_column, ondelete)
+    """
+
+    table_names: List[str] = [
+        "flavor",
+        "pipeline",
+        "pipeline_run",
+        "stack",
+        "stack_component",
+        "team_role_assignment",
+        "user_role_assignment",
+    ]
+
+    new_fk_constraints: List[Tuple[str, str, str, str, str]] = [
+        *[
+            (source, "workspace", "workspace_id", "id", "CASCADE")
+            for source in table_names
+        ],
+    ]
+    old_fk_constraints = [
+        *[
+            (source, "workspace", "project_id", "id", "CASCADE")
+            for source in table_names
+        ],
+    ]
+    return table_names, new_fk_constraints, old_fk_constraints
+
+
+def upgrade() -> None:
+    """Upgrade database schema and/or data, creating a new revision."""
+
+    table_names, new_fk_constraints, old_fk_constraints = _get_changes()
+
+    engine_name = op.get_bind().engine.name
+
+    # Under MySQL, we need to sort the old foreign keys by source table and
+    # source column first since the default foreign key names contain the
+    # foreign key number.
+    if engine_name == "mysql":
+        old_fk_constraints.sort(key=lambda x: (x[0], x[2]))
+    source_table_fk_constraint_counts: Dict[str, int] = defaultdict(int)
```

Comment:
this shouldn't be needed anymore since all the foreign keys already have an explicit name now